server/sonar-process-monitor/src/main/java/org/sonar/process/monitor/Monitor.java

```diff
@@ -282,10 +282,12 @@ private RestartorThread() {
 
     @Override
     public void run() {
+      stopProcesses();
       try {
-        stopProcesses();
         startProcesses();
       } catch (InterruptedException e) {
+        // Startup was interrupted. Processes are supposed to be stopped.
```

Comment:
that's not exactly true. Stop in case of failure inside `startProcesses` is asynchronous as handled by another thread.
So, at this point of the program, there is no garantie the processes are stopped (they are most likely not)